addons/delivery/models/sale_order.py

```diff
@@ -257,3 +163,15 @@ def _update_order_line_info(self, product_id, quantity, **kwargs):
         if self:
             self.onchange_order_line()
         return price_unit
+
+    def set_pickup_location(self, pickup_location_data):
+        """ Set the pickup location on the current sale order.
+
+        :param str pickup_location_data: The pickup location data in JSON format.
+        """
+        self.ensure_one()
+        if self.carrier_id.is_pickup and pickup_location_data:
+            pickup_location_data_json = json.loads(pickup_location_data)
+            parent_location = self.partner_id
+            address = self.env['res.partner']._address_from_json(pickup_location_data_json, parent_location, pickup_delivery_carrier_id=self.carrier_id.id)
```

Comment:
no need for `parent_location` as you use it only once and also we try to fit the code 100 symbols per line :)
```suggestion
            address = self.env['res.partner']._address_from_json(
                pickup_location_data_json,
                self.partner_id,
                pickup_delivery_carrier_id=self.carrier_id.id
            )
```